util/processinfo.go

```diff
@@ -144,5 +145,84 @@ type SessionManager interface {
 	ShowProcessList() map[uint64]*ProcessInfo
 	GetProcessInfo(id uint64) (*ProcessInfo, bool)
 	Kill(connectionID uint64, query bool)
+	KillAllConnections()
 	UpdateTLSConfig(cfg *tls.Config)
+	ServerID() uint64
+}
+
+// GlobalConnID is the global connection ID, providing UNIQUE connection IDs across the whole TiDB cluster.
+// 64 bits version:
+//  63                   41 40                                   1    0
+// +-----------------------+--------------------------------------+--------+
+// |       serverId        |             local connId             | markup |
+// |        (23b)          |                 (40b)                |(1b,==1)|
+// +-----------------------+--------------------------------------+--------+
+// 32 bits version(coming soon):
+// 31                          1    0
+// +-----------------------------+--------+
+// |             ???             | markup |
+// |             ???             |(1b,==0)|
+// +-----------------------------+--------+
+type GlobalConnID struct {
+	ServerID       uint64
+	LocalConnID    uint64
+	Is64bits       bool
+	ServerIDGetter func() uint64
+}
+
+var (
+	// MaxServerID is maximum serverID
+	MaxServerID = 1<<23 - 1
+)
+
+func (g *GlobalConnID) makeID(localConnID uint64) uint64 {
+	var (
+		id       uint64
+		serverID uint64
+	)
+	if g.ServerIDGetter != nil {
+		serverID = g.ServerIDGetter()
+	} else {
+		serverID = g.ServerID
+	}
+	if g.Is64bits {
+		id |= 0x1
+		id |= localConnID & 0xff_ffff_ffff << 1 // 40 bits local connID.
+		id |= serverID & 0x7f_ffff << 41        // 23 bits serverID.
```

Comment:
How about using `MaxServerID` here?
Is `& 0x7f_ffff ` neccessary since `localConnID` is randomed between [1, MaxServerID]?

Reply:
I design `GlobalConnID` to be a common component, so `serverID & MaxServerID` here to avoid mistakes in other occasions.